pkg/authorization/authorizer/authorizer.go

```diff
@@ -360,6 +377,48 @@ func (a DefaultAuthorizationAttributes) GetUserInfo() user.Info {
 func (a DefaultAuthorizationAttributes) GetVerb() string {
 	return a.Verb
 }
+
+// nonResourceMatches take the remainer of a URL and attempts to match it against a series of explicitly allowed steps that can end in a wildcard
+func (a DefaultAuthorizationAttributes) nonResourceMatches(rule authorizationapi.PolicyRule) bool {
+	for allowedNonResourcePath := range rule.NonResourceURLs {
+		allowPathSteps := splitPath(allowedNonResourcePath)
```

Comment:
So here's a question - why not just use strings.HasPrefix or a direct equal depending on the presence of a * in the last segment?  As you defined resource URLs you are either a path, or a path prefix.  With those rules this is a lot simpler, and frankly, simplicity here is key (the trailing slash thing we can add as a nice to have later).